Plugins/MatrixPlugin/MatrixLogic.cs

```diff
@@ -0,0 +1,334 @@
+        public List<Gear> Software;
+        public List<MatrixAction> Actions;
+
+        private int actionModifier;
+
+        private int currentActionIndex = 0;
+        public int CurrentActionIndex
+        {
+            get => currentActionIndex;
+            set
+            {
+                currentActionIndex = value;
+                OnPropertyChanged();
+                OnPropertyChanged("currentAction");
+                OnPropertyChanged("ActionDicePool");
+                OnPropertyChanged("DefenceDicePool");
+            }
+        }
+        public MatrixAction CurrentAction => Actions[currentActionIndex];
+        
+        public IHasMatrixAttributes CurrentPerson
+        {
+            get => _character.ActiveCommlink;
+            set
+            {
+                if (_character.ActiveCommlink != value)
+                {
+                    _character.ActiveCommlink = value;
+                    OnPropertyChanged();
+                    OnPropertyChanged("Attack");
+                    OnPropertyChanged("Sleaze");
+                    OnPropertyChanged("DataProcessing");
+                    OnPropertyChanged("Firewall");
+                    OnPropertyChanged("AttackMod");
+                    OnPropertyChanged("SleazeMod");
+                    OnPropertyChanged("DataProcessingMod");
+                    OnPropertyChanged("FirewallMod");
+                    OnPropertyChanged("TotalAttack");
+                    OnPropertyChanged("TotalSleaze");
+                    OnPropertyChanged("TotalDataProcessing");
+                    OnPropertyChanged("TotalFirewall");
+                }
+            }
+        }
+
+        public bool OverClocker => _character.Overclocker;
+        
+
+        public string OverClocked
+        {
+            get => CurrentPerson.Overclocked;
+            set
+            {
+                CurrentPerson.Overclocked = value;
+                OnPropertyChanged();
+                OnPropertyChanged("AttackMod");
+                OnPropertyChanged("SleazeMod");
+                OnPropertyChanged("DataProcessingMod");
+                OnPropertyChanged("FirewallMod");
+                OnPropertyChanged("TotalAttack");
+                OnPropertyChanged("TotalSleaze");
+                OnPropertyChanged("TotalDataProcessing");
+                OnPropertyChanged("TotalFirewall");
+            }
+        }
+
+        public int Attack
+        {
+            get => Parse(CurrentPerson.Attack);
+            set
+            {
+                if (CurrentPerson.Attack != value.ToString())
+                {
+                    CurrentPerson.Attack = value.ToString();
+                    OnPropertyChanged();
+                    OnPropertyChanged("TotalAttack");
+                }
+            }
+        }
+
+        public int AttackMod
+        {
+            get => Parse(CurrentPerson.ModAttack) + (CurrentPerson.Overclocked == "Attack"?1:0);
```

Comment:
I'm making this comment only for `AttackMod`, but it applies to the other Matrix attributes as well. I don't see any logic here or anywhere else where you trigger updates in these Matrix attribute values based on whether or not a character has Overclocker. In addition, the main Chummer project already does a lot of handling of Overclocker on its end to calculate total Matrix attribute values for a given `IHasMatrixAttributes`, so are you sure you are not going to end up double-counting?

Reply:
I just simplify for me Overclocker and add it to a Modifiers of Attack|Sleaze and e.t.c. I already checked it and plugin doesn't double-counting overclocker value